drivers/virtio/pci/virtio_pci.c

```diff
@@ -136,17 +135,19 @@ static int virtio_pci_handle(void *arg)
 
 	/* Reading the isr status is used to acknowledge the interrupt */
 	isr_status = virtio_cread8((void *)(unsigned long)d->pci_isr_addr, 0);
-	/* We don't support configuration interrupt on the device */
+
 	if (isr_status & VIRTIO_PCI_ISR_CONFIG) {
-		uk_pr_warn("Unsupported config change interrupt received on virtio-pci device %p\n",
-			   d);
+		/* We don't support configuration interrupt on the device */
+		uk_pr_warn("Unsupported config change interrupt received on virtio-pci device %p\n", d);
 	}
 
 	if (isr_status & VIRTIO_PCI_ISR_HAS_INTR) {
-		UK_TAILQ_FOREACH(vq, &d->vdev.vqs, next) {
+		UK_TAILQ_FOREACH(vq, &d->vdev.vqs, next)
 			rc |= virtqueue_ring_interrupt(vq);
-		}
+
+		rc = 1; /* TODO: Should not be necessary. */
```

Comment:
Agreed, since the purpose of the return value of the interrupt handlers is to let the caller know [whether to call other handlers that may be registered for this interrupt](https://github.com/unikraft/unikraft/blob/staging/plat/kvm/irq.c#L128), we can't distinguish between case where the IRQ was targeting `virtio` but for some legitimate reason there was nothing to handle (eg the virt-queue was empty), and the case where the IRQ was targeting another handler.

CC: @marcrittinghaus who is the original author of this patch, and @mschlumpp who AFAIK previously looked at the `Unhandled IRQ` message in the scope of `virtio`.